cql/src/main/java/io/stargate/cql/CqlActivator.java

```diff
@@ -57,30 +62,42 @@ private static Config makeConfig() {
   }
 
   @Override
-  public void start(BundleContext context) {
+  public void start(BundleContext context) throws InvalidSyntaxException {
     this.context = context;
     log.info("Starting CQL....");
     synchronized (cql) {
-      try {
-        context.addServiceListener(this, String.format("(Identifier=%s)", PERSISTENCE_IDENTIFIER));
-      } catch (InvalidSyntaxException ise) {
-        throw new RuntimeException(ise);
-      }
+      context.addServiceListener(
+          this,
+          String.format(
+              "(|(AuthIdentifier=%s)(Identifier=%s))", AUTH_IDENTIFIER, PERSISTENCE_IDENTIFIER));
 
-      reference = context.getServiceReference(Persistence.class.getName());
-      if (reference != null) {
-        Object service = context.getService(reference);
-        if (service != null) {
-          this.cql.start(((Persistence) service));
-          log.info("Started CQL....");
+      ServiceReference[] refs =
+          context.getServiceReferences(AuthenticationService.class.getName(), null);
+      if (refs != null) {
+        for (ServiceReference ref : refs) {
+          // Get the service object.
+          Object service = context.getService(ref);
+          if (service instanceof AuthenticationService
+              && ref.getProperty("AuthIdentifier") != null
+              && ref.getProperty("AuthIdentifier").equals(AUTH_IDENTIFIER)) {
```

Comment:
Check this out (just an idea): https://github.com/stargate/api-extensions/blob/add-sql-connection-auth/sql/server/src/main/java/io/stargate/api/sql/server/SqlActivator.java#L47

Reply:
That's awesome, thanks for the suggestion.

Reply:
@dimas-b These checks are incorrect:

Reply:
yes, I hit that already :) thanks for the pointer though :) should be fixed in the latest commit

Reply:
ah.... I forgot to push :facepalm: